### ISSUE TITLE
Configurable Elasticsearch timeout on startup similar to Mongo

### ISSUE THREAD

**Issue by knowlesc** [rwynn/monstache#168]
MongoDialSettings.Timeout allows us to extend or remove the timeout when attempting to dial mongo on startup, but no similar option exists for Elasticsearch. I'm trying to run a very simple monstache docker container and I have to manually retry after Elasticsearch starts because it's so slow. If I could configure the value sent to elastic.SetHealthcheckTimeoutStartup to be higher that would be really useful.

**Comment by rwynn** [comment]
I can look into that.  You may be able to work around this by specifying `elasticsearch-version` in your config file. Monstache pings Elasticsearch at startup to get this by default.  It might be this initial connection attempt that is failing.  If you specify it explicitly it should skip it.  

Also, you might look at the docker compose that we have which I believe is setup to wait till Elasticsearch is responding to a health check 

https://github.com/rwynn/monstache/blob/master/docker/test/docker-compose.test.yml

**Comment by knowlesc** [comment]
I did look into your docker-compose file, but unfortunately docker-compose version 3 removed the ability to depend on a healthy container. 

I tried specifying an `elasticsearch-version` but that also doesn't work, as it's failing to create an elasticsearch client on line 3622.  (`Unable to create Elasticsearch client: health check timeout: Head http://elasticsearch:9200: dial tcp 172.18.0.2:9200: connect: connection refused: no Elasticsearch node available`)

I've discovered that adding a restart policy to my docker-compose file works around this, but it is a little bit hacky. It would be nice to have monstache able to wait and retry a few times in the case that we're waiting for elasticsearch to start.
